handlers/renew.go

```
@@ -0,0 +1,75 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/coreos/etcd/client"
+	ctx "golang.org/x/net/context"
+)
+
+const (
+	// RenewTimeout is the TTL for each etcd entry
+	// Must be refreshed by etcd leader, otherwise the entry will be removed (for the failed/removed nodes)
+	RenewTimeout = 2 * 60 * time.Second
+
+	// MaxJSONSize limits the size of input Json (1MB)
+	MaxJSONSize = 1000000
+)
+
+// RenewHandler handles all PUT and GET requests related to the formation of cluster.
+// TODO: make it transactional by using etcd v3 apis
+func RenewHandler(w http.ResponseWriter, r *http.Request) {
+	// Receive list of current cluster members
+	var renewMsg RenewMsg
+	err := json.NewDecoder(r.Body).Decode(&renewMsg)
+	if err != nil {
+		fmt.Println("error: ", err)
```

need to return an http error? looks like it will continue in an error state